nifi-registry-web-api/src/main/java/org/apache/nifi/registry/web/api/AccessPolicyResource.java

```diff
@@ -174,7 +174,7 @@ public Response getAccessPolicy(
 
         final AccessPolicy accessPolicy = authorizationService.getAccessPolicy(identifier);
         if (accessPolicy == null) {
-            throw new ResourceNotFoundException("No access policy found with ID + " + identifier);
+            throw new ResourceNotFoundException("The specified access policy does not exist for this bucket.");
```

Comment:
I'm on board with the spirit of this change: removing cryptic UUIDs from the UX error messages, while still providing enough context to be useful/meaningful to the user. In general these changes look fine.

In this particular case, however, I think the wording of this message assumes too much regarding "... does not exist _for this bucket_". Access policies on the backend are generic to all types of resources, not just buckets. So for instance, one could be attempting to add a user to the access policy for the Proxy Resource, or the Tenants (users and groups) Resource. In such a case, it would be confusing to receive an error message referencing buckets.

I think we need use a more generic error message that is suitable for all cases (REST API or UI).

Reply:
Good catch....I think we can simply change it to say "The specified policy does not exist."?

Reply:
I think that works, we could even add _"...in this registry."_ to end of that. It's consistent with the other messages.